### PR TITLE
Non blocking client 1

### PR DESCRIPTION
**Pull request by k7Xm9pQr**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    Add basic structures

* what changes does this pull request make?

    Add basic structures

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

    No

### CODE REVIEW

**Review by LingKa28 — APPROVED** [review]
LGTM

File: jxline-core/src/main/java/cloud/xline/jxline/utils/Pair.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        if (o instanceof Pair<?, ?>) {
+            Pair<?, ?> e = (Pair<?, ?>) o;
+            return a.equals(e.getFirst()) && b.equals(e.getSecond());
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return a.hashCode() ^ b.hashCode();
```
Comment (bsbds) [inline jxline-core/src/main/java/cloud/xline/jxline/utils/Pair.java:35]: This should be hash both a and b.

Comment (k7Xm9pQr) [reply jxline-core/src/main/java/cloud/xline/jxline/utils/Pair.java:35]: It seems there is no such thing like `Hasher` in java.
`hashCode` is native method implemented in C/C++.

Comment (bsbds) [reply jxline-core/src/main/java/cloud/xline/jxline/utils/Pair.java:35]: Use `Objects.hash`? `a.hashCode() ^ b.hashCode()` will always result in a zero value if `a` equals `b`

Comment (k7Xm9pQr) [reply jxline-core/src/main/java/cloud/xline/jxline/utils/Pair.java:35]: We will not create a type of `Pair(a, a)`, which is similar to Map.Entry.

https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/util/Map.Entry.html#hashCode()

Comment (k7Xm9pQr) [reply jxline-core/src/main/java/cloud/xline/jxline/utils/Pair.java:35]: > Use `Objects.hash`?

A new object should not be created temporarily when calculating hash, as the resulting different hash values being calculated each time.

**Review by bsbds — APPROVED** [review]

**Pull request merged**